testsuite/gnat2goto/tests/case_expression_range/case_expression_range.adb

```diff
@@ -0,0 +1,12 @@
+procedure Case_Expression_Range is
+  Random_Val : Integer := 10;
+  Result : Integer;
+begin
+  Result := (case Random_Val is
+    when 1..10 => 1,
+    when 11..20 => 2,
```

Comment:
I suspect handling a range expression should be fairly easy - a literal range should always be translatable like this `x in a..b ===> a <= x and x <= b` I'd think